### ISSUE TITLE
🐛 BUG: Tailwind CSS classes are not working when a component has `client:* directive`

### ISSUE THREAD

**Issue by Khandakar227** [withastro/astro#4151]
### What version of `astro` are you using?

1.0.0-beta.73

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

npm

### What operating system are you using?

Linux

### Describe the Bug

Normally tailwind CSS classes work properly. I write a class name, save and it's automatically effected on the browser. However, when I use it with custom components built with other frameworks (I am using `@astrojs/svelte`) that have `client:*directive` (i.e: `client:load`) the some classes does not seem to effect the page or I should not working properly.

**Note: Classes that are written after adding `client:load` does not seem to work**

workaround: Created an npm script to build css file (according to the tailwind's official doc) and not using `@astrojs/tailwind` adapter.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-zywnbf-ccpten?file=src%2Fcomponents%2FButton.svelte

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by NotAgentBilly** [comment]
I have the same issue with astro 1.0.0-rc.6.

**Comment by matthewp** [comment]
This is an interesting one because even if you ignore the svelte component and just use `bg-blue-700` in the `index.astro` it doesn't show up. Not sure what is happening.

**Comment by matthewp** [comment]
Ah, it's because you are setting theme colors which overrides the builtin colors. You need to use extends. https://tailwindcss.com/docs/background-color#customizing-your-theme

```diff
module.exports = {
	content: ['./src/**/*.{astro,html,js,jsx,svelte,ts,tsx,vue}'],
  theme: {
-   colors: {
-     'background': '#ffbfff',
-     'primary': '#6878ff',
-     'primary-alt': '#9999ff',
-     'on-background': '#000000',
-   },
    extend: {
+      colors: {
+        'background': '#ffbfff',
+        'primary': '#6878ff',
+        'primary-alt': '#9999ff',
+        'on-background': '#000000',
+      },
    },
  },
  plugins: [],
};

```

**Comment by donaldjewkes** [comment]
@matthewp this worked for me, thanks

**Comment by Youdaman** [comment]
I've just encountered the same issue.

Minimal repro:

`components/Test.vue`
```vue
<template>
  <h1 class="text-lg">Test</h1>
</template>
```

`pages/test.astro`
```astro
---
import Test from '../components/Test.vue'
---
<h1 class="text-lg">Test</h1>
<Test client:only="vue" />
```

The above will apply the Tailwind styles, but if you remove the `h1` element from `test.astro` the styles will not be applied to the Vue component.

I also noticed that you don't even need to add an element using Tailwind classes to `test.astro` -- if you replace the `h1` element with any HTML, even nonsense like `<foo></foo>` then Tailwind kicks in for the Vue component.

So it seems that if there's _no other_ markup besides a `client:*` directive Tailwind styles do not get applied.
